Tests/UnitTests/Core/TrackFinding/CombinatorialKalmanFilterTests.cpp

```diff
@@ -22,327 +20,181 @@
-  mNavigator.resolveMaterial = true;
-  mNavigator.resolveSensitive = true;
-
-  // Use straingt line stepper to create the measurements
-  StraightLineStepper mStepper;
-
-  // Define the measurement propagator
-  using MeasurementPropagator = Propagator<StraightLineStepper, Navigator>;
-
-  // Build propagator for the measurement creation
-  MeasurementPropagator mPropagator(mStepper, mNavigator);
-
-  // Create action list for the measurement creation
-  using MeasurementActions = ActionList<MeasurementCreator>;
-  using MeasurementAborters = AbortList<EndOfWorldReached>;
-
-  auto pixelResX = Resolution(eBoundLoc0, 25_um);
-  auto pixelResY = Resolution(eBoundLoc1, 50_um);
-  auto stripResX = Resolution(eBoundLoc0, 100_um);
-  auto stripResY = Resolution(eBoundLoc1, 150_um);
-
-  ElementResolution pixelElementRes = {pixelResX, pixelResY};
-  ElementResolution stripElementResI = {stripResX};
-  ElementResolution stripElementResO = {stripResY};
+// source link selection configurtion.
+const Acts::CKFSourceLinkSelector::Config selectBestSourceLinkOnly = {
+    // global default: no chi2 cut, only one source link per surface
+    {Acts::GeometryIdentifier(), {std::numeric_limits<double>::max(), 1u}},
+};
 
-  VolumeResolution pixelVolumeRes;
-  pixelVolumeRes[2] = pixelElementRes;
-  pixelVolumeRes[4] = pixelElementRes;
+// simulation propagator
+const auto simPropagator = makeStraightPropagator(geometry);
 
-  VolumeResolution stripVolumeRes;
-  stripVolumeRes[2] = stripElementResI;
-  stripVolumeRes[4] = stripElementResO;
-  stripVolumeRes[6] = stripElementResI;
-  stripVolumeRes[8] = stripElementResO;
+// reconstruction propagator and fitter
+const auto ckfLogger =
+    Acts::getDefaultLogger("KalmanFilter", Acts::Logging::INFO);
+const auto ckfZeroPropagator = makeConstantFieldPropagator(geometry, 0_T);
+const auto ckfZero = CombinatorialKalmanFilter(ckfZeroPropagator);
 
-  DetectorResolution detRes;
-  detRes[2] = pixelVolumeRes;
-  detRes[3] = stripVolumeRes;
+std::default_random_engine rng(421235);
 
-  // Set options for propagator
-  PropagatorOptions<MeasurementActions, MeasurementAborters> mOptions(
-      tgContext, mfContext, getDummyLogger());
-  auto& mCreator = mOptions.actionList.get<MeasurementCreator>();
-  mCreator.detectorResolution = detRes;
+}  // namespace
 
-  // This vector owns the measurements
-  std::multimap<size_t, FittableMeasurement<TestSourceLink>> measurements;
-  // Make a vector of source links and further processed for KF inputs
-  std::vector<TestSourceLink> sourcelinks;
+BOOST_AUTO_TEST_SUITE(TrackFindingCombinatorialKalmanFilter)
 
-  // Set the starting positions for propagation
-  double eps = 15_mm;
-  std::map<size_t, Vector3D> startingPos;
-  startingPos.emplace(0, Vector3D{-3_m, 0., 0.});
-  startingPos.emplace(1, Vector3D{-3_m, -1.0 * eps, -1.0 * eps});
-  startingPos.emplace(2, Vector3D{-3_m, eps, eps});
+BOOST_AUTO_TEST_CASE(ZeroField) {
+  auto tracks = makeParameters();
```

Comment:
Shall we rename the `tracks` to something like `startParameters`?

Reply:
Probably better. I will fix that.

Reply:
Fixed.